engine/src/main/java/org/terasology/engine/physics/bullet/BulletPhysics.java

```diff
@@ -128,27 +161,227 @@ public BulletPhysics() {
         sequentialImpulseConstraintSolver = new btSequentialImpulseConstraintSolver();
         discreteDynamicsWorld =
                 new btDiscreteDynamicsWorld(dispatcher, broadphase, sequentialImpulseConstraintSolver, defaultCollisionConfiguration);
-        discreteDynamicsWorld.setGravity(new Vector3f(0f, -PhysicsEngine.GRAVITY, 0f));
-        blockEntityRegistry = CoreRegistry.get(BlockEntityRegistry.class);
-
+        discreteDynamicsWorld.setGravity(new Vector3f(0f, -Physics.GRAVITY, 0f));
         discreteDynamicsWorld.getBroadphase().getOverlappingPairCache().setInternalGhostPairCallback(ghostPairCallback);
+    }
 
+
+    @ReceiveEvent(components = {RigidBodyComponent.class, LocationComponent.class})
+    public void newRigidBody(OnActivatedComponent event, EntityRef entity, RigidBodyComponent rigidBody, LocationComponent location) {
```

Comment:
The methods in `PhysicsEngine` had a lot of docstrings. Would be nice to have some explanation here as well about what is happening, what the components are used for, etc.

Reply:
umm, a lot of the methods are pretty self explanatory. added a couple docstrings to clarify details.